### PR TITLE
Fix gulp: not found if it's not in PATH

### PR DESCRIPTION
**Pull request by jablko**

### CODE REVIEW

File: scripts/build/tests.js [hunk 1]
```diff
@@ -124,8 +123,8 @@ async function runConsoleTests(runJs, defaultReporter, runInParallel, watchMode,
             // finally, do a sanity check and build the compiler with the built version of itself
             log.info("Starting sanity check build...");
             // Cleanup everything except lint rules (we'll need those later and would rather not waste time rebuilding them)
-            await exec("gulp", ["clean-tsc", "clean-services", "clean-tsserver", "clean-lssl", "clean-tests"], { cancelToken });
-            const { exitCode } = await exec("gulp", ["local", "--lkg=false"], { cancelToken });
+            await exec("node", [process.argv[1], "clean-tsc", "clean-services", "clean-tsserver", "clean-lssl", "clean-tests"], { cancelToken });
```
Comment (sandersn) [inline scripts/build/tests.js]: what is `process.argv[1]` here? Is it `gulp`? How does that avoid the not found error? Is it sometimes the full path to gulp? Does the user have to provide the full path?

Comment (jablko) [reply scripts/build/tests.js]: It's the [full path to the main JavaScript file being executed](https://nodejs.org/api/process.html#process_process_argv). I double checked and I don't need to provide the full path: `process.argv[1]` is the full path when I provide a relative path, e.g.
```Shell
> CI=true relative/gulp runtests-parallel
```
or
```Shell
> CI=true node relative/gulp runtests-parallel
```

**Comment by sandersn** [comment]
I just noticed that this code path executes only on CI, which is supposed to have gulp on the path. Why is this change useful? Wouldn't we want to know if CI is set up incorrectly?

**Comment by sandersn** [comment]
This PR is getting stale, so I'm going to close it to keep the number of open PRs manageable. @jablko if you want to continue work on it, the next step would be to address whether the change is useful on CI.

**Pull request closed without merging**